gigascience/gigadb-website
# Fix accessibility and styling issues in admin FUW workflow

# Pull request for issue: #1696

This PR should be merged after https://github.com/gigascience/gigadb-website/pull/1691

This is a pull request for the following functionalities:

* Fix a11y issues and styling issues related to FUW workflow in pages:
  * http://gigadb.gigasciencejournal.com/adminDataset/admin

## How to test?

* follow FUW workflow up to the point where admin clicks on dropbox icon
* Review styles for action buttons at the top of the page
![image](https://github.com/gigascience/gigadb-website/assets/143437854/df1f5465-f614-41b3-a5ed-1d70bf9855fd)
* use keyboard to Interact with the "customize instructions" button to open the modal
![image](https://github.com/gigascience/gigadb-website/assets/143437854/87f870f9-95ed-4fb1-a95f-c6c6f0207c49)

* Use tab key within modal to verify that focus is trapped inside and causes no scrolling
* Click on "close" button and verify that focus is returned to button that opened modal
* Open modal again, enter any value and save
* Verify that focus is returned to button that opened modal again

## How have functionalities been implemented?

* Update styles of dropbox success panel

* NOTE:
  * Opted to not change the styles of the success alert (which were reported in the related issue) as considered this is a more general change that would involve updating all alerts (and similar elements) on the site. I added some considerations about this here: https://github.com/gigascience/gigadb-website/issues/1696#issuecomment-1951860611
![image](https://github.com/gigascience/gigadb-website/assets/143437854/fec6f83a-211b-4307-b001-ad61ac326065)

* Update custom instructions modal
  * Fix spacing and layout issues
  * Update modal button styles
  * When user clicks on "save changes" and modal closes, focus is restored to modal trigger button
  * Correctly handle modal focus trap (when pressing the tab key, focus cycles within the modal, bootstrap should be able to do it, but I observe a strange behavior (page scrolling and an invisible element getting focused) and thus the fix, which leverages an existing script (i.e. protected/js/trap-focus.js)

## Any issues with implementation?

One of the issues reported in the ticket seem backend-related. More information in the ticket itself: https://github.com/gigascience/gigadb-website/issues/1696#issue-2141382592

## Any changes to automated tests?

* One failing acceptance test, but I think it might be brittle or false failure because it's unrelated to these changes and the test passes on second run: ✖ A user visit gigadb website: Go to the Jobs page from the main page (8.31s)

## Any changes to documentation?

--

## Any technical debt repayment?

--

## Any improvements to CI/CD pipeline?

--
